### ISSUE TITLE
17.10.0 Introduced a Breaking Change.

### ISSUE THREAD

**Issue by kevinmstephens** [terraform-aws-modules/terraform-aws-eks#1570]
## Description

We saw builds breaking 2 hours ago and I found this change that I think caused it and the timing matches up.
```
Terraform v1.0.3
on linux_amd64
Initializing plugins and modules...
╷
│ Error: Unsupported block type
│ 
│   on .terraform/modules/eks_cluster.eks/modules/node_groups/node_groups.tf line 72, in resource "aws_eks_node_group" "workers":
│   72:   dynamic "update_config" {
│ 
│ Blocks of type "update_config" are not expected here.
╵
Operation failed: failed running terraform plan (exit 1)
```

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

 - Terraform: 1.0.3

 - Provider(s):   required_providers {
    aws = {
      source = "hashicorp/aws"
    }
    datadog = {
      source = "DataDog/datadog"
    }
    kubernetes = {
      source  = "hashicorp/kubernetes"
      version = ">= 2.0.0"
    }
    rancher2 = {
      source = "rancher/rancher2"
    }
    spotinst = {
      source = "spotinst/spotinst"
    }
    mysql = {
      source = "terraform-providers/mysql"
    }
    github = {
      source = "integrations/github"
    }
  }

 - Module: "terraform-aws-modules/eks/aws" 17.9.0 last known working version. 17.10.0 showed error.

**Comment by bryantbiggs** [comment]
Can you please provide some additional information to help troubleshoot where the issue is emanating from

```
Versions

- Terraform:

- Provider(s):

- Module:
```

**Comment by antonbabenko** [comment]
@kevinmstephens thanks for opening this issue!

My bad, I missed the point that `update_config` support was added in [v3.56.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#3560-august-26-2021) of Terraform AWS provider which was released last week.

I will make another release to fix versions requirements now.

**Comment by antonbabenko** [comment]
@kevinmstephens Fixed in #1571. [v17.11.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/v17.11.0) has been just released. Please try it and open a new issue if there is anything wrong.

I apologize for the incomplete release in v17.10.0.
